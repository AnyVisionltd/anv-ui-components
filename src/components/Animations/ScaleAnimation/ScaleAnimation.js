import React from 'react'
import propTypes from 'prop-types'
import { CSSTransition } from 'react-transition-group'
import './ScaleAnimation.scss'

const ScaleAnimation = ({
  children,
  isOpen,
  verticalStart,
  horizontalStart,
  ...otherProps
}) => (
  <CSSTransition
    in={isOpen}
    unmountOnExit
    classNames={`scale-${verticalStart}-${horizontalStart}`}
    timeout={200}
    {...otherProps}
  >
    {children}
  </CSSTransition>
)

ScaleAnimation.defaultProps = {
  verticalStart: 'center',
  horizontalStart: 'center',
}

ScaleAnimation.propTypes = {
  /** If <code>true</code> display <code>children</code>. */
  isOpen: propTypes.bool.isRequired,
  /** Animation vertical starting position. */
  verticalStart: propTypes.oneOf(['center', 'top', 'bottom']),
  /** Animation vertical starting position. */
  horizontalStart: propTypes.oneOf(['center', 'start', 'end']),
  /** The component content. */
  children: propTypes.element.isRequired,
}

export default ScaleAnimation
