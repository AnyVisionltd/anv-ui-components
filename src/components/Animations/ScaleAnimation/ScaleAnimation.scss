@mixin scale($fromVertical, $fromHorizontal) {
  transform-origin: $fromVertical $fromHorizontal;
  &-enter {
    opacity: 0;
    transform: scale(0.8);
    transform-origin: $fromVertical $fromHorizontal;
  }
  &-enter-active {
    opacity: 1;
    transform: scale(1);
    transition: opacity 200ms, transform 200ms;
    transform-origin: $fromVertical $fromHorizontal;
  }
  &-exit {
    opacity: 1;
    transform-origin: $fromVertical $fromHorizontal;
  }
  &-exit-active {
    opacity: 0;
    transform: scale(0.8);
    transform-origin: $fromVertical $fromHorizontal;
    transition: opacity 200ms, transform 200ms;
  }
}

.scale {
  &-center-center {
    @include scale(center, center)
  }

  &-top {
    &-start {
      @include scale(top, left)
    }

    &-end {
      @include scale(top, right)
    }
  }

  &-bottom {
    &-start {
      @include scale(bottom, left)
    }

    &-end {
      @include scale(bottom, right)
    }
  }
}

html[dir=rtl] {
  .scale {
    &-top {
      &-start {
        @include scale(top, right)
      }

      &-end {
        @include scale(top, left)
      }
    }

    &-bottom {
      &-start {
        @include scale(bottom, right)
      }

      &-end {
        @include scale(left, left)
      }
    }
  }
}