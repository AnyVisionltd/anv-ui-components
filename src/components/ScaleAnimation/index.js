export { default as Scale } from './ScaleAnimation'
