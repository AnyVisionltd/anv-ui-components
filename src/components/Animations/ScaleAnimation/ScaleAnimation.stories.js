import React, { useState } from 'react'
import ScaleAnimation from './ScaleAnimation'
import { Button } from '../../Button'
import { centerDecorator } from '../../../utils/storybook/decorators'
import styles from '../../../storybook/index.module.scss'

export default {
  title: 'Animations/Scale',
  component: ScaleAnimation,
  decorators: [centerDecorator],
}

export const Basic = () => {
  const [scaleConfig, setScaleConfig] = useState({
    isOpen: false,
    verticalStart: 'top',
    horizontalStart: 'start',
  })
  const { isOpen, verticalStart, horizontalStart } = scaleConfig
  const scale = (vertical = 'center', horizontal = 'center') => {
    setScaleConfig({
      isOpen: !isOpen,
      verticalStart: vertical,
      horizontalStart: horizontal,
    })
  }

  return (
    <div style={ {
      height: '250px', display: 'flex', flexDirection: 'column', alignItems: 'center',
    } }
    >
      <div className={ styles.marginFlexContainer }>
        <Button onClick={ () => scale('top', 'start') }>
          start from top, left to right
        </Button>
        <Button onClick={ () => scale('top') }>
          start from top, center
        </Button>
        <Button onClick={ () => scale('top', 'end') }>
          start from top, right to left
        </Button>
      </div>
      <div className={ styles.marginFlexContainer }>
        <Button onClick={ () => scale('center', 'start') }>
          start from the middle, left to right
        </Button>
        <Button onClick={ () => scale() }>
          pop from center
        </Button>
        <Button onClick={ () => scale('center', 'end') }>
          start from middle, right to left
        </Button>
      </div>
      <div className={ styles.marginFlexContainer }>
        <Button onClick={ () => scale('bottom', 'start') }>
          start from bottom, left to right
        </Button>
        <Button onClick={ () => scale('bottom') }>
          start from bottom, center
        </Button>
        <Button onClick={ () => scale('bottom', 'end') }>
          start from bottom, right to left
        </Button>
      </div>
      <div className={ styles.marginFlexContainer }>
        <ScaleAnimation
          isOpen={ isOpen }
          horizontalStart={ horizontalStart }
          verticalStart={ verticalStart }
        >
          <h2 style={ { textAlign: 'center' } }>Scaled text</h2>
        </ScaleAnimation>
      </div>
    </div>
  )
}
