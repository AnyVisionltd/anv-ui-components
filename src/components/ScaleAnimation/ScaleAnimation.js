import React from 'react'
import propTypes from 'prop-types'
import { CSSTransition } from 'react-transition-group'
import './ScaleAnimation.scss'

const ScaleAnimation = ({
  children, isOpen, verticalStart, horizontalStart, ...otherProps
}) => (
  <CSSTransition
    in={ isOpen }
    unmountOnExit
    classNames={ `scale-${verticalStart}-${horizontalStart}` }
    timeout={ 200 }
    { ...otherProps }
  >
    { children }
  </CSSTransition>
)

ScaleAnimation.defaultProps = {
  verticalStart: 'top',
  horizontalStart: 'start',
}

ScaleAnimation.propTypes = {
  /** If <code>true</code> display <code>children</code>. */
  isOpen: propTypes.bool.isRequired,
  /** Animation vertical starting position. */
  verticalStart: propTypes.oneOf(['top', 'bottom']),
  /** Animation vertical starting position. */
  horizontalStart: propTypes.oneOf(['start', 'end']),
  /** The component content. */
  children: propTypes.element.isRequired,
}

export default ScaleAnimation
