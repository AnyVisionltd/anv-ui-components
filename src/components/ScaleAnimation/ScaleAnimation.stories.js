import React, { useState } from 'react'
import ScaleAnimation from './ScaleAnimation'
import { Button } from '../../index'
import { centerDecorator } from '../../utils/storybook/decorators'
import styles from '../../styles/storybook/index.module.scss'

export default {
  title: 'Animations/ScaleAnimation',
  component: ScaleAnimation,
  decorators: [centerDecorator],
}

export const Basic = () => {
  const [slide, setSlide] = useState({ isOpen: false, direction: 'up' })

  return (
    <div>
      <div className={ styles.marginFlexContainer }>
        <Button onClick={ () => setSlide({ isOpen: !slide.isOpen, direction: 'up' }) }>up</Button>
        <ScaleAnimation isOpen={ slide.isOpen }>
          <h2 style={ { textAlign: 'center' } }>Slider</h2>
        </ScaleAnimation>
      </div>
    </div>
  )
}
