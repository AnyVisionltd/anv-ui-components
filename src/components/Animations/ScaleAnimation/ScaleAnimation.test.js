import React from 'react'
import { render } from '@testing-library/react'
import ScaleAnimation from './ScaleAnimation'

describe('<ScaleAnimation/>', () => {
	it('should render when isOpen true', () => {
		const { container } = render(<ScaleAnimation isOpen><div>test</div></ScaleAnimation>)
		expect(container).toMatchSnapshot()
	})

	it('should NOT render when isOpen false', () => {
		const { container } = render(<ScaleAnimation isOpen={ false }><div>test</div></ScaleAnimation>)
		expect(container).toMatchSnapshot()
	})
})
