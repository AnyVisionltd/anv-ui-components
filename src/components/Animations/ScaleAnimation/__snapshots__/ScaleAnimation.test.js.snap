// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ScaleAnimation/> should NOT render when isOpen false 1`] = `<div />`;

exports[`<ScaleAnimation/> should render when isOpen true 1`] = `
<div>
  <div>
    test
  </div>
</div>
`;
